TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -113,6 +113,41 @@ private void Enter(Node node, CodeElement context = null, SymbolTable table = nu
 
             if (node.CodeElement != null)
                 NodeCodeElementLinkers.Add(node.CodeElement, node);
+
+            DataDefinition parent = node.Parent as DataDefinition;
+            DataDescriptionEntry dataEntry = node.CodeElement as DataDescriptionEntry;
+
+            if (parent != null && dataEntry != null)
+            {
+                DataDescriptionEntry parentDataEntry = parent.CodeElement as DataDescriptionEntry;
+
+                if (parentDataEntry != null)
+                {
+                    if (parent.IsSynchronized)
+                    {
+                        dataEntry.IsSynchronized = parentDataEntry.IsSynchronized;
```

Comment:
We're not supposed to update CodeElement during Node phase.
In incremental mode outside the modified text are not parsed/updated.
Which means, if you set property IsSynchronized on a CodeElement, it will stay on the next incremental phase even if the CobolNodeBuilder doesn't update it another time.

I propose the following behavior:
- properties on CodeElement are tight to the source code
  - if the SYNC keyword is used on a data declaration, then the property is set on the CodeElement
- properties on Node are related to CodeElement or their parent
   - if a parent node have the IsSynchronized property to true, then the current Node will also have the IsSynchronized property to true